realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -232,224 +244,252 @@ private void emitConstructor(JavaWriter writer) throws IOException {
-                writer.emitStatement("links.clear()");
-                writer.beginControlFlow("if (value == null)");
-                    writer.emitStatement("return");
-                writer.endControlFlow();
-                writer.beginControlFlow("for (RealmModel linkedObject : (RealmList<? extends RealmModel>) value)");
-                    writer.beginControlFlow("if (!(RealmObject.isManaged(linkedObject) && RealmObject.isValid(linkedObject)))");
-                        writer.emitStatement("throw new IllegalArgumentException(\"Each element of 'value' must be a valid managed object.\")");
-                    writer.endControlFlow();
-                    writer.beginControlFlow("if (((RealmObjectProxy)linkedObject).realmGet$proxyState().getRealm$realm() != proxyState.getRealm$realm())");
-                        writer.emitStatement("throw new IllegalArgumentException(\"Each element of 'value' must belong to the same Realm.\")");
-                    writer.endControlFlow();
-                    writer.emitStatement("links.add(((RealmObjectProxy)linkedObject).realmGet$proxyState().getRow$realm().getIndex())");
-                writer.endControlFlow();
-                writer.endMethod();
-            } else {
-                throw new UnsupportedOperationException(
-                        String.format("Type '%s' of field '%s' is not supported", fieldTypeCanonicalName, fieldName));
+                // set value as default value
+                writer.emitStatement("final Row row = proxyState.getRow$realm()");
+                writer.beginControlFlow("if (value == null)")
+                    .emitSingleLineComment("Table#nullifyLink() does not support default value. Just using Row.")
+                    .emitStatement("row.nullifyLink(%s)", fieldIndexVariableReference(field))
+                    .emitStatement("return")
+                    .endControlFlow();
+                writer.beginControlFlow("if (!RealmObject.isValid(value))")
+                    .emitStatement("throw new IllegalArgumentException(\"'value' is not a valid managed object.\")")
+                    .endControlFlow();
+                writer.beginControlFlow("if (((RealmObjectProxy) value).realmGet$proxyState().getRealm$realm() != proxyState.getRealm$realm())")
+                    .emitStatement("throw new IllegalArgumentException(\"'value' belongs to a different Realm.\")")
+                    .endControlFlow();
+                writer.emitStatement("row.getTable().setLink(%s, row.getIndex(), ((RealmObjectProxy) value).realmGet$proxyState().getRow$realm().getIndex(), true)",
+                    fieldIndexVariableReference(field));
+                writer.emitStatement("return");
             }
-            writer.emitEmptyLine();
-        }
+        });
+        writer.emitStatement("proxyState.getRealm$realm().checkIfValid()");
```

Comment:
It works in some places.  When there are control structures in the source, as well as the emitted code, it just gets weird.  I'll put this back the way it was.